app/Scripts/boid.gd

```diff
@@ -298,11 +299,16 @@ func _on_AwakenBoidTrigger_area_entered(area):
 	#if area.is_in_group("Hitbox") and area.get_parent().has_meta("Enemy"):
 	#	damage_boid(50)
 
-func add_damage(value: int) -> void:
+func add_damage(value: int, knockback_dealer: Node) -> void:
 	health -= value
 	health_calculation()
+
+
+	if _enemy_manager._enemies.has(knockback_dealer):
```

Comment:
instead of using enemy_manager i think it would be preferable to just check if knockback_dealer instance is valid